### ISSUE TITLE
Stops sending logs after an error

### ISSUE THREAD

**Issue by andrewbaxter** [kislyuk/watchtower#46]
I'm not 100% sure this is an issue in watchtower, but AFAICT after sending a log message that was too large (530+kb, larger than 256kb limit) I stopped seeing logs appearing in the cloudwatch log group.  Truncating some large fields fixed the issue.

I just wanted to leave a note here in case someone has the chance to look into it.  If we stop getting logs we're pretty much blind so I think an investigation is important.

**Comment by andrewbaxter** [comment]
Here's the error:
```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib64/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/.../lib/python3.6/site-packages/watchtower/__init__.py", line 156, in batch_sender
    self._submit_batch(cur_batch, stream_name)
  File "/home/.../lib/python3.6/site-packages/watchtower/__init__.py", line 92, in _submit_batch
    response = self.cwl_client.put_log_events(**kwargs)
  File "/home/.../lib/python3.6/site-packages/botocore/client.py", line 253, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/.../lib/python3.6/site-packages/botocore/client.py", line 543, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.InvalidParameterException: An error occurred (InvalidParameterException) when calling the PutLogEvents operation: Log event too large: 521983 bytes exceeds limit of 262144
```

**Comment by andrewbaxter** [comment]
Just taking a quick look, it looks like this would escape https://github.com/kislyuk/watchtower/blob/master/watchtower/__init__.py#L108 .  I think there's should probably be a inspecific `except:` somewhere so no unhandled exceptions accidentally kill the thread.

**Comment by kislyuk** [comment]
Yes, I've observed this type of behavior too. Thanks for noting, I agree uncaught errors probably shouldn't kill the logger thread. PRs welcome, I might get around to fixing this and instrumenting a test on the weekend.

**Comment by bowman** [comment]
I think I've seen this but due to a different exception:

```
[..]/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
[..]/site-packages/watchtower/__init__.py", line 156, in batch_sender
[..]  self._submit_batch(cur_batch, stream_name)
[..]/site-packages/watchtower/__init__.py", line 92, in _submit_batch
[..]   response = self.cwl_client.put_log_events(**kwargs)
[..]/site-packages/botocore/client.py", line 253, in _api_call
[..]     return self._make_api_call(operation_name, kwargs)
...
[..]site-packages/botocore/retryhandler.py", line 359, in _check_caught_exception
SSLError: EOF occurred in violation of protocol (_ssl.c:661)
```
(The EOF is probably only some transient network issue)

The effect is that certain streams drop out, no longer publishing to CloudWatch,
while others continue.  Restarting fixed the issue.

I'm also concerned that messages may be queueing up indefinitely

Killing the thread might be ok, as long as the dead one is cleaned-up.

**Comment by jcerjak** [comment]
Was this fixed in https://github.com/kislyuk/watchtower/commit/718cf88322f8ba84d3f53b54224aed1e3f04400e ?

**Comment by kislyuk** [comment]
This is now fixed in v2.0.0 by catching and forwarding all errors to the logging system.
